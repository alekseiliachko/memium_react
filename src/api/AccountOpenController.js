import APIController from "./ApiController";

class АccountOpenController extends APIController {
  constructor() {
    super();
  }

  async getAccountsDetails(accountId) {
    const details = await this.request(
      "get",
      `/open/account/details/${accountId}`
    );
    return details;
  }

  async getAccountsAvatar(accountId) {
    const avatar = await this.request(
      "get",
      `/open/account/avatar/${accountId}`,
      {
        responseType: "blob",
      }
    );
    return URL.createObjectURL(avatar.data);
  }
}

export default new АccountOpenController();
