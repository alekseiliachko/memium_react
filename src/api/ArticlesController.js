import APIController from "./ApiController";

class AccountsController extends APIController {
  constructor() {
    super();
  }

  async createArticle(articleData) {
    return await this.request("post", "article/", articleData).then(
      (val) => val.data
    );
  }

  async updateArticle(newArticleData) {
    return await this.request("put", "article/", newArticleData);
  }

  async articleHasLike(articleId) {
    return await this.request("get", `article/haslike/${articleId}`);
  }

  async setImageForArticle(articleId, image) {
    return await this.request("post", `article/image/${articleId}`, image).then(
      (val) => val.data
    );
  }

  async myArticles() {
    return await this.request("get", "article/my");
  }

  async deleteArticle(articleId) {
    return await this.request("delete", `article/my/${articleId}`);
  }

  async getArticleById(articleId) {
    return await this.request("get", `/open/article/articleId/${articleId}`);
  }

  async getArticleImage(articleId) {
    const image = await this.request(
      "get",
      `/open/article/image/${articleId}`,
      {
        responseType: "blob",
      }
    );
    if (!image.data.size) {
      return null;
    }
    return URL.createObjectURL(image.data);
  }
}

export default new AccountsController();
