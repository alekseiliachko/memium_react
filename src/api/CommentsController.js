import APIController from "./ApiController";

class CommentsController extends APIController {
  constructor() {
    super();
  }

  async createComment(articleId, content) {
    const somewhat = await this.request("post", "/comment/", {
      articleId,
      content,
    });
    return somewhat.data;
  }

  async deleteComment(commentId) {
    const somewhat = await this.request("delete", `/comment/${commentId}`);
    return somewhat.data;
  }

  async getComments(articleId) {
    const somewhat = await this.request("get", `/open/comment/${articleId}`);
    return somewhat.data;
  }
}

export default new CommentsController();
