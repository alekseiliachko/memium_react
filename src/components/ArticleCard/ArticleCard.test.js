import React from "react";
import { MemoryRouter } from "react-router";
import { createMount } from "@material-ui/core/test-utils";
import { ArticleCard } from "./index";
import ArticlesController from "../../api/ArticlesController";
import CommentsController from "../../api/CommentsController";
import AccountOpenController from "../../api/AccountOpenController";
jest.mock("../../api/ArticlesController");
jest.mock("../../api/AccountOpenController");
jest.mock("../../api/CommentsController");

const props = {
  data: {
    id: "5b28e638-ef12-45a7-b629-84350cc0c92e",
    imageUrl: null,
    authorId: "5d48bb9e-f1d9-41d2-bb02-9c4f578bd282",
    title: "Bio?",
    description: null,
    date: 1615047293904,
    category: "IT",
  },
  onLike: jest.fn(),
  onUnLike: jest.fn(),
  liked: true,
};

describe("article view page", () => {
  it("renders correctly", async () => {
    ArticlesController.getArticleImage.mockResolvedValue("");
    AccountOpenController.getAccountsDetails.mockResolvedValue({
      data: { name: "" },
    });
    CommentsController.getComments.mockResolvedValue([]);

    const renderedValue = createMount()(
      <MemoryRouter>
        <ArticleCard {...props} />
      </MemoryRouter>
    );

    expect(renderedValue.html()).toMatchSnapshot();
  });
});
