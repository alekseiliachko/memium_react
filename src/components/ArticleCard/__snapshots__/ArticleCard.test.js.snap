// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`article card renders correctly 1`] = `"<div class=\\"MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-elevation1 MuiPaper-rounded\\"><div class=\\"MuiBox-root MuiBox-root-4\\"><div class=\\"MuiBox-root MuiBox-root-5\\"><h4 class=\\"MuiTypography-root MuiTypography-h4\\">Bio?</h4><div class=\\"MuiBox-root MuiBox-root-6\\"><h6 class=\\"MuiTypography-root makeStyles-wrap-3 MuiTypography-subtitle1 MuiTypography-colorTextSecondary\\"></h6></div><div class=\\"MuiBox-root MuiBox-root-7\\"><h6 class=\\"MuiTypography-root MuiTypography-subtitle1\\"></h6><div><button class=\\"MuiButtonBase-root MuiIconButton-root\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiIconButton-label\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M12 21.35l-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54L12 21.35z\\"></path></svg></span><span class=\\"MuiTouchRipple-root\\"></span></button></div></div><h6 class=\\"MuiTypography-root MuiTypography-subtitle2 MuiTypography-colorTextSecondary\\">Sat Mar 06 2021</h6></div><div class=\\"MuiBox-root MuiBox-root-8\\"><div class=\\"MuiAvatar-root MuiAvatar-square makeStyles-articleImage-2 MuiAvatar-colorDefault\\"><svg class=\\"MuiSvgIcon-root MuiAvatar-fallback\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z\\"></path></svg></div></div></div></div>"`;
